lib/xcknife/stream_parser.rb

```diff
@@ -88,24 +92,27 @@ def compute_shards_for_partitions(test_time_for_partitions)
     end
 
     def test_time_for_partitions(historical_events, current_events = nil)
-      analyzer =  EventsAnalyzer.for(current_events, relevant_partitions)
+      analyzer = EventsAnalyzer.for(current_events, relevant_partitions)
       @stats[:current_total_tests] = analyzer.total_tests
       times_for_target_class = Hash.new { |h, current_target| h[current_target] = Hash.new(0) }
       each_test_event(historical_events) do |target_name, result|
         next unless relevant_partitions.include?(target_name)
+
         inc_stat :historical_total_tests
-        next unless analyzer.has_test_class?(target_name, result.className)
+        next unless analyzer.test_class?(target_name, result.className)
+
         times_for_target_class[target_name][result.className] += (result.totalDuration * 1000).ceil
       end
 
       extrapolate_times_for_current_events(analyzer, times_for_target_class) if current_events
       hash_partitions(times_for_target_class)
     end
 
+    # rubocop:disable Metrics/CyclomaticComplexity
```

Comment:
Alternative to this is configuring the threshold higher. Not sure how far off this method is compared to the configuration.

Reply:
In some cases it was really high (21/7), I wouldn't want to bump it in the rules as I think seeing these exceptions is more actionable. I think the right thing is to decompose the logic in follow-up PRs.